### PR TITLE
Adding support for Redis6 config and new features

### PR DESCRIPTION
**Pull request by dom-nie**
#### Pull Request (PR) description
This commit introduces the latest config for Redis6 and Sentinel, with new parameters. Additionally, the MemoryPolicy type is introduced with some small adjustments in the rest of the code.

As discussed previously on voxpupuli IRC this is a base commit, and depending on the review I will make changes accordingly.

### CODE REVIEW

File: templates/redis6-sentinel.conf.erb [hunk 1]
```diff
@@ -0,0 +1,334 @@
+# *** IMPORTANT ***
```
Comment (bastelfreak) [inline templates/redis6-sentinel.conf.erb:1]: can you please convert the template to epp? We would like to get rid of erb on the long run and to achieve this, all new templates have to be epp.

File: REFERENCE.md [hunk 2]
```diff
@@ -42,849 +43,1033 @@ repositories.
 
 ## Classes
 
-### `redis`
+### <a name="redis"></a>`redis`
```
Comment (ekohl) [inline REFERENCE.md:46]: Could you regenerate `REFERENCE.md` prior to the changes in a separate commit to make reviewing easier?

File: spec/classes/redis_spec.rb [hunk 3]
```diff
@@ -679,7 +679,7 @@ class { 'redis::globals':
 
         it do
           if facts[:operatingsystem] == 'Ubuntu' && facts[:operatingsystemmajrelease] == '16.04'
-            is_expected.not_to contain_file(config_file_orig).with_content(%r{protected-mode})
+            is_expected.not_to contain_file(config_file_orig).with_content(%r{^protected-mode$})
```
Comment (ekohl) [inline spec/classes/redis_spec.rb:682]: The regex isn't really needed. In fact, I'd say it's better not to be strict. It's expecting that `protected-mode` isn't present at all, regardless of where in the file it shows up.

File: spec/classes/redis_spec.rb [hunk 4]
```diff
@@ -43,8 +43,8 @@
 
         it do
           is_expected.to contain_service(service_name).with(
-            'ensure'     => 'running',
-            'enable'     => 'true',
+            'ensure' => 'running',
+            'enable' => 'true',
```
Comment (ekohl) [inline spec/classes/redis_spec.rb:47]: Generally it makes it easier to review if you separate out the visual changes to a separate commit.

File: templates/redis6-sentinel.conf.erb (hunk 1 above)
Comment (dom-nie) [reply templates/redis6-sentinel.conf.erb]: @bastelfreak Done

File: REFERENCE.md (hunk 2 above)
Comment (dom-nie) [reply REFERENCE.md:46]: @ekohl I did the split to two commits

File: spec/classes/redis_spec.rb (hunk 3 above)
Comment (dom-nie) [reply spec/classes/redis_spec.rb:670]: I would say it's better to be strict.
If we only set variables, without touching documentation inside the file, so people can use them for reference then 
`# protected-mode example` would be shown somewhere which is caught by the previous regexp and the test fails

File: spec/classes/redis_spec.rb (hunk 4 above)
Comment (dom-nie) [reply spec/classes/redis_spec.rb]: Removed from the scope of this changes

**Comment by ekohl** [comment]
This now has a merge conflict.
